WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/CustomRangeTabCreationCoordinator.swift

```diff
@@ -18,18 +26,35 @@ final class CustomRangeTabCreationCoordinator: Coordinator {
 
 private extension CustomRangeTabCreationCoordinator {
     func presentDateRangePicker() {
+        let buttonTitle = (startDate == nil && endDate == nil) ? Localization.add : nil
+        let endDate = self.endDate ?? Date()
+        let startDate = self.startDate ?? Date(timeInterval: -Constants.thirtyDaysInSeconds, since: endDate) // 30 days before end date
         let controller = RangedDatePickerHostingController(
+            startDate: startDate,
+            endDate: endDate,
             datesFormatter: DatesFormatter(),
-            customApplyButtonTitle: Localization.add,
+            customApplyButtonTitle: buttonTitle,
             datesSelected: { [weak self] start, end in
                 guard let self else { return }
+                guard start < end else {
+                    self.presentInvalidTimeRangeAlert()
+                    return
+                }
                 self.onDateRangeSelected(start, end)
             }
         )
 
         navigationController.present(controller, animated: true)
     }
 
+    func presentInvalidTimeRangeAlert() {
+        let alert = UIAlertController(title: Localization.InvalidTimeRangeAlert.title,
+                                      message: Localization.InvalidTimeRangeAlert.message,
+                                      preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: Localization.InvalidTimeRangeAlert.action, style: .cancel))
+        navigationController.present(alert, animated: true)
```

Comment:
An improvement to this would be to display the alert without the picker being dismissed, (since if there's an error, a user would likely want to pick a new date right away).

But, I don't think it's a blocker for this PR.

Reply:
Nice observation! I see that the same issue happens when selecting an invalid date range in Analytics Hub, so I've moved the validation to the date picker to be reused for both flows in eafe4dbe2fda1e84825a4d12bf895b045ac51cbc.